lib/api/core.dart

```diff
@@ -198,6 +199,13 @@ Map<String, String> authHeader({required String email, required String apiKey})
   };
 }
 
+// TODO(#453): Create real User-Agent string
+Map<String, String> userAgentHeader() {
+  return {
+    'User-Agent': 'ZulipMobile/?.?.? (Android 14)',
```

Comment:
Let's work in something here to make clear this is actually the Flutter app, not the RN app. E.g., saying "ZulipFlutter".

Tim suggests at https://github.com/zulip/zulip-flutter/issues/453#issuecomment-1856146265 that it's enough for "ZulipMobile" to appear somewhere in the value, not necessarily at the start. Can we add something like "(like ZulipMobile)" at the end instead, and put "ZulipFlutter" at the start?

Or if we do need it at the start, in the venerable tradition of every modern browser claiming to be Mozilla 5.0:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent
then in that same venerable tradition I hope we can at least put the more informative name somewhere later.

Reply:
Tested with live server and dug through https://github.com/zulip/zulip/blob/main/zerver/lib/user_agent.py . The User-Agent is parsed with this:
```python
pattern = re.compile(
    """^ (?P<name> [^/ ]* [^0-9/(]* )
    (/ (?P<version> [^/ ]* ))?
    ([ /] .*)?
    $""",
    re.X,
)
```

and the whitelist from https://github.com/zulip/zulip/blob/main/zerver/models/clients.py is:
```python
            sending_client
            in (
                "zulipandroid",
                "zulipios",
                "zulipdesktop",
                "zulipmobile",
                "zulipelectron",
                "zulipterminal",
                "snipe",
                "website",
                "ios",
                "android",
            )
            or "desktop app" in sending_client
            # Since the vast majority of messages are sent by humans
            # in Zulip, treat test suite messages as such.
            or (sending_client == "test suite" and settings.TEST_SUITE)
```

To trigger a match we could not start the string with "ZulipFlutter". I think given the options we should present as "ZulipMobile" and stick on a "(like ZulipMobile)" at the end.

We could possibly start with "ZulipFlutter; like desktop app" or "ZulipFlutter(like desktop app)" (but NOT "ZulipFlutter (like desktop app)") but I don't like these options.

Reply:
OK, let's do a string like `ZulipMobile/1.2.3 (Android 14) (actually ZulipFlutter)`. Where instead of "Android 14" we have the actual OS name and version (and we can leave that part out for a first pass), and instead of "1.2.3" we have the actual version (and we can leave that out too for a first pass).

Reply:
In fact, better yet: once we do have the version number, it'll be from a different sequence from the zulip-mobile version numbers anyway. So rather than saying "ZulipMobile/1.2.3", we can let the version number remain absent at that spot and instead say a string like `ZulipMobile (Android 14) (actually ZulipFlutter/1.2.3)`.